streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionJoinProcessorSupplier.java

```diff
@@ -31,53 +30,44 @@
+    private final KTableValueGetterSupplier<KRight, VRight> foreignValueGetterSupplier;
 
-    public SubscriptionJoinProcessorSupplier(final KTableValueGetterSupplier<KO, VO> foreignValueGetterSupplier) {
+    public SubscriptionJoinProcessorSupplier(final KTableValueGetterSupplier<KRight, VRight> foreignValueGetterSupplier) {
         this.foreignValueGetterSupplier = foreignValueGetterSupplier;
     }
 
     @Override
-    public Processor<CombinedKey<KO, K>, Change<ValueAndTimestamp<SubscriptionWrapper<K>>>, K, SubscriptionResponseWrapper<VO>> get() {
-
-        return new ContextualProcessor<CombinedKey<KO, K>, Change<ValueAndTimestamp<SubscriptionWrapper<K>>>, K, SubscriptionResponseWrapper<VO>>() {
-
-            private KTableValueGetter<KO, VO> foreignValues;
+    public Processor<CombinedKey<KRight, KLeft>, Change<ValueAndTimestamp<SubscriptionWrapper<KLeft>>>, KLeft, SubscriptionResponseWrapper<VRight>> get() {
+        return new ContextualProcessor<>() {
+            private KTableValueGetter<KRight, VRight> foreignValues;
 
             @Override
-            public void init(final ProcessorContext<K, SubscriptionResponseWrapper<VO>> context) {
+            public void init(final ProcessorContext<KLeft, SubscriptionResponseWrapper<VRight>> context) {
                 super.init(context);
                 foreignValues = foreignValueGetterSupplier.get();
                 foreignValues.init(context);
             }
 
             @Override
-            public void process(final Record<CombinedKey<KO, K>, Change<ValueAndTimestamp<SubscriptionWrapper<K>>>> record) {
+            public void process(final Record<CombinedKey<KRight, KLeft>, Change<ValueAndTimestamp<SubscriptionWrapper<KLeft>>>> record) {
                 Objects.requireNonNull(record.key(), "This processor should never see a null key.");
                 Objects.requireNonNull(record.value(), "This processor should never see a null value.");
-                final ValueAndTimestamp<SubscriptionWrapper<K>> valueAndTimestamp = record.value().newValue;
+                final ValueAndTimestamp<SubscriptionWrapper<KLeft>> valueAndTimestamp = record.value().newValue;
                 Objects.requireNonNull(valueAndTimestamp, "This processor should never see a null newValue.");
-                final SubscriptionWrapper<K> value = valueAndTimestamp.value();
-
-                if (value.version() > SubscriptionWrapper.CURRENT_VERSION) {
-                    //Guard against modifications to SubscriptionWrapper. Need to ensure that there is compatibility
-                    //with previous versions to enable rolling upgrades. Must develop a strategy for upgrading
-                    //from older SubscriptionWrapper versions to newer versions.
-                    throw new UnsupportedVersionException("SubscriptionWrapper is of an incompatible version.");
-                }
+                final SubscriptionWrapper<KLeft> value = subscriptionWrapper(valueAndTimestamp);
```

Comment:
Small refactoring proposed by IntelliJ -- guess it make sense to extract into a helper method -- easier to read overall, as semantics are a little clearer.